scfw/verifiers/osv_verifier/osv_advisory.py

```diff
@@ -0,0 +1,212 @@
+"""
+Provides a representation of OSV advisories for use in `OsvVerifier`.
+"""
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from cvss import CVSS2, CVSS3, CVSS4  # type: ignore
+
+
+class Severity(Enum):
+    """
+    Represents the possible severities that an OSV advisory can have.
+    """
+    Non = 0
+    Low = 1
+    Medium = 2
+    High = 3
+    Critical = 4
+
+    def __lt__(self, other: "Severity") -> bool:
+        """
+        Compare two `Severity` instances.
+
+        Args:
+            self: The `Severity` to be compared on the left-hand side
+            other: The `Severity` to be compared on the right-hand side
+
+        Returns:
+            A `bool` indicating whether `<` holds between the two given `Severity`.
+
+        Raises:
+            TypeError: The other argument given was not a `Severity`.
+        """
+        if self.__class__ is not other.__class__:
+            raise TypeError(
+                f"'<' not supported between instances of '{self.__class__}' and '{other.__class__}'"
+            )
+
+        return self.value < other.value
+
+    def __str__(self) -> str:
+        """
+        Format a `Severity` for printing.
+
+        Returns:
+            A `str` representing the given `Severity` suitable for printing.
+        """
+        return "None" if self.name == "Non" else self.name
```

Comment:
Aint better to use the value in the if? 

```suggestion
        return "None" if self.value == self.Non else self.name
```

Also, why not just change the enum: Non -> None

Reply:
```python
>>> from scfw.verifiers.osv_verifier.osv_advisory import Severity
Traceback (most recent call last):
...
    None = 0
    ^^^^
SyntaxError: cannot assign to None
```

Tbh this shouldn't be an issue because this None is inside the Severity namespace. 🤷

Reply:
I can use the value in the return statement.

Reply:
oh, right, is a reserved word

Reply:
After thinking on this one a bit more, I think I will leave it the way it is using `self.name` as the criterion.  The variant names are defined in the CVSS standard and thus won't be changed whereas the numbers are arbitrary.  This implementation also communicates intent slightly better (i.e., I wanted it to be `self.name` but `None` is reserved, so I have to do this thing).